voter-dapp/src/ActiveRequests.js

```diff
@@ -263,10 +281,38 @@ function ActiveRequests({ votingAccount, votingGateway }) {
         encryptedVote
       });
       indicesCommitted.push(index);
+
+      // Store price and salt that we will attempt to encrypt on-chain in a cookie. This way, if the encryption
+      // (or subsequent decryption) fails, then the user can still recover their committed price and salt and
+      // reveal their commit manually. Note that this will store a cookie for each call to `encryptMessage`, even
+      // if the user never signs and submits the `batchCommitFunction` to commit their vote on-chain. Therefore,
+      // the cookies could store more hash data than the user ends up committing on-chain. It is the user's
+      // responsibility to determine which commit data to use.
+      const newCommitKey = toVotingAccountAndPriceRequestKey(
+        votingAccount,
+        pendingRequests[index].identifier,
+        pendingRequests[index].time
+      );
+      const updatedCommitBackups = Object.assign(
+        {},
+        {
+          ...cookies[newCommitKey],
+          [Date.now()]: {
```

Comment:
nit: maybe compute `Date.now()` outside the loop, so all the commits for a particular button-click have the same timestamp?

Reply:
done